### ISSUE TITLE
React-Navigation custom TabNavigation component hides header

### ISSUE THREAD

**Issue by apamaffe** [react-community/react-navigation#477]
**TLDR:** Can i customise the view passed to const _TabNav = TabNavigator({});_, or is there a way to display the headers of nested stack navigators when the tab navigator is created with _createNavigationContainer(createNavigator(CustomTabRouter)(CustomTabView));_

I'm trying to build a custom view component that replaces the standard tab bar view (based on a floating action button).

I've looked at the StacksInTabs example which works great, but i don't see a way to customise the tabview.

When following the CustomTabs example in the docs i can't get the nested StackNavigator to display the header.

Something is definitely different when you create the TabNavigator manually by passing a router object and a view object to createNavigationContainer, but i can't work out what.

Here's a couple of samples on rnplay.org:

Using the default nav creation functions: https://rnplay.org/apps/n0ixXw Composing my own Tab Navigator: https://rnplay.org/apps/tzTecA

They don't run on the site due to requiring outside module imports, but here's my package.json

{ "name": "SampleApp", "version": "0.0.1", "private": true, "scripts": { "start": "node node_modules/react-native/local-cli/cli.js start", "test": "jest" }, "dependencies": { "react": "15.4.2", "react-native": "0.41.0", "react-native-action-button": "^2.0.22", "react-native-vector-icons": "^4.0.0", "react-navigation": "^1.0.0-beta.3" }, "devDependencies": { "babel-jest": "18.0.0", "babel-preset-react-native": "1.9.1", "jest": "18.1.0", "react-test-renderer": "15.4.2" }, "jest": { "preset": "react-native" } }

**Comment by serg2007** [comment]
have same problem

**Comment by emusgrave** [comment]
See [#1686] for possible solution

**Comment by spencercarli** [comment]
Gonna close for inactivity (cleaning up issues) and it seems there may be a solution in #1686
